### SYGMA web interface
Loading graphical user interface ... this takes a moment. This user interface is still being tested. Please report bugs to critter@uvic.ca.

In [1]:
start_SYGMA()

Box(children=(HBox(children=(Tab(children=(VBox(children=(HBox(children=(Text(value=u'1.0', description=u'Total stellar mass [$M_{\\odot}$]:', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible', width=u'14em'), style=DescriptionStyle(description_width=u'initial')), Dropdown(description=u'Initial metallicity: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible', width=u'14em'), options=('0.02', '0.01', '0.006', '0.001', '0.0001', '0.0'), style=DescriptionStyle(description_width=u'initial'), value='0.02')), layout=Layout(border=u'0em none', margin=u' 0em 0em 0em 0em', visibility=u'visible')), HBox(children=(Text(value=u'1.0e10', description=u'Final time [yr]: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible', width=u'14em'), style=DescriptionStyle(description_width=u'initial')), Text(value=u'1.0e7', description=u'Time step [yr]: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible', width=u'14em'), style=DescriptionStyle(description_width=u'initial'))), layout=Layout(border=u'0em none', margin=u' 0em 0em 0em 0em', visibility=u'visible')), HBox(children=(Dropdown(description=u'IMF type: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible', width=u'14em'), options=('salpeter', 'chabrier', 'kroupa', 'alphaimf'), style=DescriptionStyle(description_width=u'initial'), value='salpeter'), FloatSlider(value=0.0, description=u'Set alpha: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'hidden'), max=5.0)), layout=Layout(border=u'0em none', margin=u' 0em 0em 0em 0em', visibility=u'visible')), HBox(children=(Text(value=u'1.0', description=u'IMF lower limit [$M_{\\odot}$]: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible', width=u'14em'), style=DescriptionStyle(description_width=u'initial')), Text(value=u'30.0', description=u'IMF upper limit [$M_{\\odot}$]: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible', width=u'14em'), style=DescriptionStyle(description_width=u'initial'))), layout=Layout(border=u'0em none', margin=u' 0em 0em 0em 0em', visibility=u'visible')), HBox(children=(Checkbox(value=True, description=u'Include SNe Ia: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible')), Dropdown(description=u'SNe Ia rates: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible', width=u'14em'), options=('Power law', 'Exponential', 'Gaussian', 'Maoz12'), style=DescriptionStyle(description_width=u'initial'), value='Power law')), layout=Layout(border=u'0em none', margin=u' 0em 0em 0em 0em', visibility=u'visible')), HBox(children=(ToggleButtons(description=u'CCSN remnant prescription:', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible'), options=('Analytic prescription', 'Ye=0.4982'), value='Analytic prescription'), Dropdown(layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible', width=u'14em'), options={'Delay': 'yield_tables/agb_and_massive_stars_nugrid_MESAonly_fryer12delay.txt', 'Mix': 'yield_tables/agb_and_massive_stars_nugrid_MESAonly_fryer12mix.txt', 'Rapid': 'yield_tables/agb_and_massive_stars_nugrid_MESAonly_fryer12rapid.txt'}, style=DescriptionStyle(description_width=u'initial'), value='yield_tables/agb_and_massive_stars_nugrid_MESAonly_fryer12delay.txt')), layout=Layout(border=u'0em none', margin=u' 0em 0em 0em 0em', visibility=u'visible')), HBox(children=(Button(description=u'Run simulation', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible'), style=ButtonStyle(font_weight=u'bold')), Button(description=u'Remove selected', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visibility=u'visible'), style=ButtonStyle(font_weight=u'bold')), Text(value=u'', description=u'Run name: ', layout=Layout(border=u'0.5em LightGrey', margin=u'0.25em', visi

### Simulation Parameter Definitions
| Parameter     | Definition   |
| ------------- | -------------|
| `Total stellar mass` | Mass of the initial star-forming cloud in solar masses. All mass is transformed into stars. |
| `Initial metallicity` | Metal fraction $Z$ of the initial star-forming cloud. | 
| `Final time` | Total time of SSP evolution in years. When the final time is reached the code stops.|
| `Time step` | Size of the initial time step in years. Afterwards time step sizes increase logarithmically. |
| `IMF type` | Choose type of the initial mass function. Setting `alphaimf` allows setting the negative exponent of a single power law. If a custom IMF was defined via the tab menu it can be selected. |
| `IMF lower limit` & `IMF upper limit` | Initial mass range of the initial mass function. Stellar yields are applied between $1\, {\rm M}_\odot$ and $30\, {\rm M}_\odot$. |
| `Include SNIe Ia` | Option to exclude any contribution from supernova (SN) Ia. |
| `SNe Ia rates` | Choice of delay-time distribution for SN Ia. |
| `CCSN remnant prescription` | Remnant mass prescription used for the core-collapse supernova (CCSN) explosion (Affects composition of stellar yields).|
| | `Analytic prescription` let you choose form the mass-and metallicity dependent fallback prescriptions of [Fryer 2012](http://adsabs.harvard.edu/abs/2012ApJ...749...91F) . |
| | `Ye=0.4982` uses a fallback coordinate at a fixed electron fraction Ye=0.4982. |

References:


### Known Issues
* If `IMF lower limit` and `IMF upper limit` are unreasonable close together a error is thrown.

In [2]:
from IPython.display import HTML

HTML('''<script>
code_show=true; 
function code_toggle() {
 if (code_show){
 $('div.input').hide();
 } else {
 $('div.input').show();
 }
 code_show = !code_show
} 
$( document ).ready(code_toggle);
</script>
<form action="javascript:code_toggle()"><input type="submit" value="Toggle Code"></form>''')

